### PR TITLE
Update observability-driven testing commands in documentation

### PR DESCRIPTION
**Pull request by johnnydacosta**
Replaced the `make start-odd` and related Docker command in the "Enable API observability-driven testing" section with updated commands (`make run-tracetestings` and `docker compose -f docker-compose-tests.yml run traceBasedTests`) to reflect current available commands. The previous `make start-odd` command no longer exists.

### CODE REVIEW

**Review by svrnm — APPROVED** [review]

File: content/en/docs/demo/docker-deployment.md [hunk 1]
```diff
@@ -50,13 +50,13 @@ docker compose up --force-recreate --remove-orphans --detach
     {{< tabpane text=true >}} {{% tab Make %}}
 
 ```shell
-make start-odd
+make run-tracetestings
```
Comment (julianocosta89) [inline content/en/docs/demo/docker-deployment.md:53]:
```suggestion
make run-tracetesting
```
https://github.com/open-telemetry/opentelemetry-demo/blob/main/Makefile#L107

File: content/en/docs/demo/docker-deployment.md [hunk 2]
```diff
@@ -69,7 +69,7 @@ Once the images are built and containers are started you can access:
 - Grafana: <http://localhost:8080/grafana/>
 - Load Generator UI: <http://localhost:8080/loadgen/>
 - Jaeger UI: <http://localhost:8080/jaeger/ui/>
-- Tracetest UI: <http://localhost:11633/>, only when using `make start-odd`
+- Tracetest UI: <http://localhost:11633/>, only when using `make run-tracetestings`
```
Comment (julianocosta89) [inline content/en/docs/demo/docker-deployment.md:72]:
```suggestion
- Tracetest UI: <http://localhost:11633/>, only when using `make run-tracetesting`
```

**Review by julianocosta89 — CHANGES_REQUESTED** [review]
Hello @johnnydacosta thx for fixing it.
Would you mind applying my suggestions?
The make command is actually named: `run-tracetesting`.

**Comment by johnnydacosta** [comment]
Hi @julianocosta89, I just applied the fix now :)

**Review by julianocosta89 — APPROVED** [review]
Thank you!
LGTM

**Pull request merged**